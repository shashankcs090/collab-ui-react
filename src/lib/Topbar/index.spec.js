import React from 'react';
import { shallow, mount } from 'enzyme';
import { Topbar } from '@collab-ui/react';

describe('<Topbar />', () => {
  it('should render a Topbar', () => {
    const wrapper = shallow(<Topbar />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle brandAnchorElement prop', () => {
    const container = mount(<Topbar brandAnchorElement={<div className='testAnchor' />}/>);

    expect(container.find('.testAnchor').length).toEqual(1);
  });

  // it('should render a Topbar with AppTitle' , () => {
  //   const wrapper = mount(<Topbar appTitle='Test' />);
  //
  //   expect(wrapper.props().appTitle).toEqual('Test');
  // });
});
// import renderer from 'react-test-renderer';
// import {shallow, mount} from 'enzyme';
// import Checkbox from '../Checkbox';
//
// describe('<Checkbox />', () => {
// 	/**
// 	 * Snapshot based testing for look and feel.
// 	 */
// 	it('should render regular default checkbox', () => {
// 		const rendered = renderer.create(
// 			<Checkbox value="sample_12" label="Sample Checkbox" />
// 		);
// 		expect(rendered.toJSON()).toMatchSnapshot();
// 	});
//
// 	it('should render regular checked checkbox', () => {
// 		const rendered = renderer.create(
// 			<Checkbox value="sample_12" label="Sample Checkbox" checked />
// 		);
// 		expect(rendered.toJSON()).toMatchSnapshot();
// 	});
//
// 	it('should render disabled checkbox', () => {
// 		const rendered = renderer.create(
// 			<Checkbox value="sample_12" label="Sample Checkbox" disabled />
// 		);
// 		expect(rendered.toJSON()).toMatchSnapshot();
// 	});
//
// 	it('should render reversed checkbox', () => {
// 		const rendered = renderer.create(
// 			<Checkbox value="sample_12" label="Sample Checkbox" reverse />
// 		);
// 		expect(rendered.toJSON()).toMatchSnapshot();
// 	});
//
// 	it('should render an indeterminate state checkbox', () => {
// 		const rendered = renderer.create(
// 			<Checkbox value="sample_12" label="Sample Checkbox" indeterminate />
// 		);
// 		expect(rendered.toJSON()).toMatchSnapshot();
// 	});
//
// 	it('should have a `disabled` attribute when `disabled` is `true`', () => {
// 		const wrapper = mount(
// 			<Checkbox value="sample_12" label="Sample Checkbox" disabled />
// 		);
// 		expect(wrapper.find('Checkbox').find('input').html().includes('disabled')).toBe(true);
// 	});
//
// 	it('should render with additional classes applied', () => {
// 		const rendered = renderer.create(
// 			<Checkbox value="sample_12" label="Sample Checkbox" className="additional" />
// 		);
// 		expect(rendered.toJSON()).toMatchSnapshot();
// 	});
//
// 	it('should render the element with any extra attributes that are passed', () => {
// 		const rendered = renderer.create(
// 			<Checkbox value="sample_12" label="Sample Checkbox" aria-label="Checkbox" />
// 		);
// 		expect(rendered.toJSON()).toMatchSnapshot();
// 	});
//
// 	it('should apply custom id if passed', () => {
// 		const rendered = renderer.create(
// 			<Checkbox value="sample_12" label="Sample Checkbox" id="mycheck" />
// 		);
// 		expect(rendered.toJSON()).toMatchSnapshot();
// 	});
//
// 	/**
// 	 * Functionality testing. This single test case covers the sanity for checkbox behavior.
// 	 * jest.fn() mocks a function.
// 	 * Refer: https://facebook.github.io/jest/docs/mock-functions.html
// 	 */
// 	it('should be clickable ', () => {
// 		const onCheckboxClick = jest.fn();
// 		const wrapper = shallow(<Checkbox value="sample_1234" onClick={onCheckboxClick} />);
// 		wrapper.first().simulate('click');
// 		expect(onCheckboxClick).toHaveBeenCalled();
// 	});
//
// 	it('should get checked with a single click', () => {
// 		const wrapper = mount(<Checkbox value="sample_12" label="Sample Checkbox" />);
// 		wrapper.find('input').simulate('change', { target: { value: 'on' } });
// 		expect(wrapper.find('svg').hasClass('opacity0')).toBe(false);
// 	});
//
// 	it('should get unchecked when clicked twice', () => {
// 		const wrapper = mount(<Checkbox value="sample_12" label="Sample Checkbox" />);
// 		wrapper.find('input').simulate('change', { target: { value: 'on' } });
// 		wrapper.find('input').simulate('change', { target: { value: 'off' } });
// 		expect(wrapper.find('svg').hasClass('opacity0')).toBe(true);
// 	});
//
// 	it('should call the user defined onChange when passed', () => {
// 		const onCheckChange = jest.fn();
// 		const wrapper = mount(<Checkbox value="sample_12" label="Sample Checkbox" onChange={onCheckChange} />);
// 		wrapper.find('Checkbox').first().find('input').simulate('change');
// 		expect(onCheckChange).toHaveBeenCalled();
// 	});
// });
