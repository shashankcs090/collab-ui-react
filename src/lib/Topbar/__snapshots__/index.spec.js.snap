// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Topbar /> should render a Topbar 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Topbar
    anchor={null}
    brandAnchorElement={null}
    className=""
    color="dark"
    fixed={false}
    icon="icon-cisco-logo"
    image={null}
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="cui-top-bar__container row"
      >
        <div
          className="cui-top-bar__brand"
        >
          <a
            className="cui-brand"
            href={null}
          >
            <div
              className="cui-brand__logo"
            >
              <i
                className="icon icon-cisco-logo"
              />
            </div>
            <div
              className="cui-brand__title"
            >
              
            </div>
          </a>
        </div>
      </div>,
      "className": "cui-top-bar cui-top-bar--dark",
      "role": "navigation",
    },
    "ref": [Function],
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="cui-top-bar__brand"
          >
            <a
              className="cui-brand"
              href={null}
            >
              <div
                className="cui-brand__logo"
              >
                <i
                  className="icon icon-cisco-logo"
                />
              </div>
              <div
                className="cui-brand__title"
              >
                
              </div>
            </a>
          </div>,
          null,
        ],
        "className": "cui-top-bar__container row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="cui-brand"
              href={null}
            >
              <div
                className="cui-brand__logo"
              >
                <i
                  className="icon icon-cisco-logo"
                />
              </div>
              <div
                className="cui-brand__title"
              >
                
              </div>
            </a>,
            "className": "cui-top-bar__brand",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="cui-brand__logo"
                >
                  <i
                    className="icon icon-cisco-logo"
                  />
                </div>,
                <div
                  className="cui-brand__title"
                >
                  
                </div>,
              ],
              "className": "cui-brand",
              "href": null,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "cui-brand__logo",
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="icon icon-cisco-logo"
                  />,
                  "className": "cui-brand__logo",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon icon-cisco-logo",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": "cui-brand__title",
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": "cui-brand__title",
                },
                "ref": null,
                "rendered": "",
                "type": "div",
              },
            ],
            "type": "a",
          },
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="cui-top-bar__container row"
        >
          <div
            className="cui-top-bar__brand"
          >
            <a
              className="cui-brand"
              href={null}
            >
              <div
                className="cui-brand__logo"
              >
                <i
                  className="icon icon-cisco-logo"
                />
              </div>
              <div
                className="cui-brand__title"
              >
                
              </div>
            </a>
          </div>
        </div>,
        "className": "cui-top-bar cui-top-bar--dark",
        "role": "navigation",
      },
      "ref": [Function],
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="cui-top-bar__brand"
            >
              <a
                className="cui-brand"
                href={null}
              >
                <div
                  className="cui-brand__logo"
                >
                  <i
                    className="icon icon-cisco-logo"
                  />
                </div>
                <div
                  className="cui-brand__title"
                >
                  
                </div>
              </a>
            </div>,
            null,
          ],
          "className": "cui-top-bar__container row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="cui-brand"
                href={null}
              >
                <div
                  className="cui-brand__logo"
                >
                  <i
                    className="icon icon-cisco-logo"
                  />
                </div>
                <div
                  className="cui-brand__title"
                >
                  
                </div>
              </a>,
              "className": "cui-top-bar__brand",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="cui-brand__logo"
                  >
                    <i
                      className="icon icon-cisco-logo"
                    />
                  </div>,
                  <div
                    className="cui-brand__title"
                  >
                    
                  </div>,
                ],
                "className": "cui-brand",
                "href": null,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "cui-brand__logo",
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      className="icon icon-cisco-logo"
                    />,
                    "className": "cui-brand__logo",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "icon icon-cisco-logo",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": "cui-brand__title",
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "className": "cui-brand__title",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "div",
                },
              ],
              "type": "a",
            },
            "type": "div",
          },
          null,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
